plans/subplans/install.pp

```diff
@@ -135,6 +135,20 @@
   # either be undef or else the key content to write.
   $r10k_private_key = peadm::file_or_content('r10k_private_key', $r10k_private_key_file, $r10k_private_key_content)
 
+  # enable code manager if:
+  # * it isn't explicitly disabled *and* the user provided r10k repo+key
+  # * a replica is present
+  # * one or multiple compiler are present
+  $_code_manager_auto_configure = if $r10k_private_key and $code_manager_auto_configure {
```

Comment:
This conditional doesn't account for the case where a user might provide just a remote and not a key, as they would if they were using an HTTPS clone URL for a public repo, which doesn't require any creds or key. Would it make sense to check for the remote to be set here instead of the key, since I think that is the minimum required for Code Manager to work properly?

Reply:
derp, you're right! I had it the way you described it before and broke it during the latest rebase...

Reply:
Fixed it in https://github.com/puppetlabs/puppetlabs-peadm/pull/401